decoimpact/business/entities/rules/depth_average_rule.py

```diff
@@ -48,20 +38,14 @@ def execute(
         """
 
         # The first DataArray in our value_arrays contains the values to be averaged
-        # but the name of the key is given by the user, and is unknown here, so
-        # just use the first value.
-        variables = next(iter(value_arrays.values()))
-        interface_suffix = _get_layer_suffix(self._layer_type, logger)
+        # but the name of the key is given by the user, and is unknown here, so use
+        # the ordering defined in the parser.
+        values_list = list(value_arrays.values())
 
-        bed_level_values = _extract_variable_based_on_suffix(
-            value_arrays, BED_LEVEL_SUFFIX
-        )
-        depths_interfaces = _extract_variable_based_on_suffix(
-            value_arrays, interface_suffix
-        )
-        water_level_values = _extract_variable_based_on_suffix(
-            value_arrays, WATER_LEVEL_SUFFIX
-        )
+        variables = values_list[0]
+        bed_level_values = values_list[1]
```

Comment:
Should "variables" be plural? You are getting a single element from the list or is that element a list in itself?

Reply:
agreed, changed to variable